### ISSUE TITLE
Add option to trim "query string" part of the copied URLs

### ISSUE THREAD

**Issue by yphastos** [k08045kk/CopyTabTitleUrl#75]
Sometimes the urls contain a query string part with a lot of useless information for the end user.
they are usually "tracking" parameters, to know where did you access the url from (an email, campaign, ad), etc.

e.g. if I open a url from my reddit newsletter's email, the url looks like:

```
Star Wars Outlaws PC System Requirements : r/nvidia
https://www.reddit.com/r/nvidia/comments/1eh66sa/star_wars_outlaws_pc_system_requirements/?%24deep_link=true&correlation_id=40b361cb-ae29-4c51-a521-305cbe585b46&post_fullname=t3_1eh66sa&post_index=2&ref=email_digest&ref_campaign=email_digest&ref_source=email&utm_content=post_title&%243p=e_as&utm_medium=Email%20Amazon%20SES
```

in which the whole "query string" part (everything after the first '?') is useless for me.

I often like to copy the urls of interesting posts, to share them including the title (for which precisely I use your extension) but the extra parameters are not needed, they only make the link convoluted, too long, and ugly.

so I'd like the extension to have an option to omit all those "useless" parameters (the whole query string in this case), and just copy the essential part of the URL, e.g.

```
Star Wars Outlaws PC System Requirements : r/nvidia
https://www.reddit.com/r/nvidia/comments/1eh66sa/star_wars_outlaws_pc_system_requirements/
```

I know some sites rely on the query string parameters for actually telling the site what info we need, and give users a specific page, but for a lot of cases it is completely irrelevant for the end user (like the example from reddit).

So, since it MAY be useful sometimes, I think this setting should be OPTIONAL to NOT copy those query strings. (and the default should be to copy the whole URL "as is").

e.g. perhaps each time we open the extensions "popup" to choose if we want title, url or both, it could have a checkbox to indicate whether we want the query string part or not.

I hope you can implement something. Thanks

Edit:
I know it is a minor inconvenience, and that I could just edit the copied url in whatever I paste it, but it would be nice to have it implemented directly.

**Comment by k08045kk** [comment]
I understood the following.


"I want to copy the URL with the query string removed."
Or,
"I want to copy the URL with the query string and fragment identifier removed."
(The fragment identifier is any string after the "#")


Try the following: 

1. Enable [Use Extended Mode]
2. Specify the following format in [Format] > format

	${origin}${pathname}${hash}
	${origin}${pathname}

3. Check the operation on the page with the query string


Remark
For more information on Format, see the following page
https://github.com/k08045kk/CopyTabTitleUrl/wiki/Format
https://github.com/k08045kk/CopyTabTitleUrl/wiki/Format#url


That's all.

**Comment by yphastos** [comment]
Perfect!

I didn't know it was possible to use specific parts of the URL in the format!

The options are a bit confusing, specially because only 2 formats can be shown in the popup (I haven't used them before), as opposed to the context menu option, with up to 9 formats, but I managed to do it.

What I did was assigning the following:

${title}${enter}${origin}${pathname}

To the format (#1) field, and now I can copy the URLs exactly as I wanted! thanks!

And the best part is that I can keep the default options, and also use my desired format to manually choose it for reddit or similar sites.

Thank you very much!

I'm closing this as solved.

**Comment by yphastos** [comment]
Perfect!

I didn't know it was possible to use specific parts of the URL in the format!

The options are a bit confusing, specially because only 2 formats can be shown in the popup (I haven't used them before), as opposed to the context menu option, with up to 9 formats, but I managed to do it.

What I did was assigning the following:

${title}${enter}${origin}${pathname}

To the format (#1) field, and now I can copy the URLs exactly as I wanted! thanks!

And the best part is that I can keep the default options, and also use my desired format to manually choose it for reddit or similar sites.

Thank you very much!

I'm closing this as solved.
